### DISCUSSION TITLE
On Mac, I cannot add my SSH Key to the keychain

### DISCUSSION THREAD

**Discussion by jchansen925** [community/community#72299] [New to GitHub]
Hi, I apologize if this has been answered before. I could not find an answer. I have generated a new SSH key and want to add it to my private keychain, however, I keep getting an error saying "No Such File or Directory" after following the steps to generate the SSH key and create and update the .ssh/config file.  My steps in the terminal are as follows: 

`Your public key has been saved in /Users/johnhansen/.ssh/id_ALGORITHM.pub
The key fingerprint is:
SHA256:vL0NwVP/GY2NotzH+9d7dfUZrd44j2jrgZPSVyh/Lrc john.hansenux@gmail.com
The key's randomart image is:
+--[ED25519 256]--+
|                 |
|                 |
|            .   .|
|       . . . o =+|
|        S = o =oB|
|         = X +.o*|
|        o X =.++=|
|         . *o*=o=|
|          .o=+E**|
+----[SHA256]-----+
johnhansen@Johns-MacBook-Pro ~ % ssh-add --apple-use-keychain ~/.ssh/id_ed25519
/Users/johnhansen/.ssh/id_ed25519: No such file or directory`

I don't know how to get past this error. Any help would be greatly appreciated. 
Thanks, 
John

**Comment by Xrenya** [comment]
I think you save with different name and you are trying to add another name, test it: `ssh-add --apple-use-keychain /Users/johnhansen/.ssh/id_ALGORITHM.pub `

**Reply by jchansen925** [reply]
Thanks Xyrenya. That got me pass the last errors above, but now I get this error:  git clone git@github.com:jchansen925/ton.git
Cloning into 'ton'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights and the repository exists.
johnhansen@Johns-MacBook-Pro ~ % 

I tried following the instructions on this page: 
https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey

But I still get the same error.

**Reply by jchansen925** [reply]
This issue has been resolved. Thanks Xyrenya!

John
